### ISSUE TITLE
Incompatible with Go 1.23, which doesn't allow //go:linkname to internal symbols, including to runtime package 

### ISSUE THREAD

**Issue by stefanb** [pingcap/tidb#54197]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Try building with Go 1.23 rc1, eg in while testing Go 1.23 rc1 in homebrew:
* https://github.com/Homebrew/homebrew-core/pull/175310

### 2. What did you expect to see? (Required)

Succesfull build

### 3. What did you see instead (Required)

Build error:
````
  # github.com/pingcap/tidb/dumpling/cmd/dumpling
  link: github.com/pingcap/tidb/pkg/util/printer: invalid reference to runtime.buildVersion
````

### 4. What is your TiDB version? (Required)

----

Note, that Go 1.23 no longer allows `//go:linkname * runtime.*` link instructioins, see 
* https://tip.golang.org/doc/go1.23#linker
* https://github.com/golang/go/issues/67401

It is used several times in tidb:
https://github.com/search?q=repo%3Apingcap%2Ftidb%20linkname&type=code

**Comment by hawkingrei** [comment]
Great, Thank for your report. We will start to support the go1.23. TiDB 8.5 will use go 1.23.

**Comment by rsc** [comment]
Note that you can use runtime/debug.ReadBuildInfo and then use info.GoVersion instead of runtime.buildVersion.

**Comment by hawkingrei** [comment]
It has been fixed by #53619.

I have success to build tidb with go1.23rc1.

**Comment by lance6716** [comment]
> It has been fixed by #53619.
> 
> I have success to build tidb with go1.23rc1.

the original post is about dumpling not tidb. Can you check make build for dumpling?

**Comment by stefanb** [comment]
Pleas don't forget to tag a release sometime before Go 1.23 is released in August.
